cms/djangoapps/maintenance/tests.py

```diff
@@ -0,0 +1,267 @@
+        return course
+
+    @ddt.data(
+        ('', COURSE_KEY_ERROR_MESSAGES['empty_course_key']),
+        ('edx', COURSE_KEY_ERROR_MESSAGES['invalid_course_key']),
+        ('course-v1:e+d+X', COURSE_KEY_ERROR_MESSAGES['course_key_not_found']),
+    )
+    @ddt.unpack
+    def test_invalid_course_key_messages(self, course_key, error_message):
+        """
+        Test all error messages for invalid course keys.
+        """
+        # validate that course key contains error message
+        self.verify_error_message(
+            data={'course-id': course_key},
+            error_message=error_message
+        )
+
+    def test_non_split_course(self):
+        """
+        Test that we get a error message on non-split courses.
+        """
+        # validate non split error message
+        course = CourseFactory.create(default_store=ModuleStoreEnum.Type.mongo)
+        self.verify_error_message(
+            data={'course-id': unicode(course.id)},
+            error_message='Force publishing course is not supported with old mongo courses.'
+        )
+
+    def test_already_published(self):
+        """
+        Test that when a course is forcefully publish, we get a 'course is already published' message.
+        """
+        course = self.setup_test_course()
+        # force publish the course
+        response = self.get_force_publish_course_response(course, is_dry_run=False)
+        self.validate_success_from_response(response, 'Forced published the course')
+
+        # now course is forcefully published, we should get already published course.
+        self.verify_error_message(
+            data={'course-id': unicode(course.id), 'dry-run': ''},
+            error_message='Course is already in published state.'
+        )
+
+    def verify_versions_are_different(self, course):
+        """
+        Verify draft and published versions point to different locations.
+
+        Arguments:
+            course (object): a course object.
+        """
+        # get draft and publish branch versions
+        versions = get_course_versions(unicode(course.id))
+
+        # verify that draft and publish point to different versions
+        self.assertNotEqual(versions['draft-branch'], versions['published-branch'])
+
+    def get_force_publish_course_response(self, course, is_dry_run=True):
+        """
+        Get force publish the course response.
+
+        Arguments:
+            course (object): a course object.
+            is_dry_run (bool): dry run force publish course, default is True.
+
+        Returns:
+            response (HttpResponse): response from force publish post view.
+        """
+        # Verify versions point to different locations initially
+        self.verify_versions_are_different(course)
+
+        # force publish course view
+        data = {
+            'course-id': unicode(course.id),
+            'dry-run': 'on' if is_dry_run else ''
```

Comment:
Oh, interesting. I had no idea that's what happened when you submitted a POST. Do you mind adding a comment to that effect?